geomstats/geometry/stratified/graph_space.py

```diff
@@ -381,141 +571,474 @@ def dist(self, graph_a, graph_b):
+        self._n_nodes = None
+
+    def _set_all_perms(self, metric):
+        n_nodes = metric.n_nodes
+        if self._all_perms is None or self._n_nodes != n_nodes:
+            self._n_nodes = n_nodes
+            self._all_perms = gs.array(
+                list(itertools.permutations(range(n_nodes), n_nodes))
+            )
+
+    def _align_single(self, metric, base_graph, graph_to_permute):
+        permuted_graphs = metric.space.permute(graph_to_permute, self._all_perms)
+        dists = metric.total_space_metric.dist(base_graph, permuted_graphs)
+        return self._all_perms[gs.argmin(dists)]
+
+    def align(self, metric, base_graph, graph_to_permute):
+        """Align graphs.
+
+        Parameters
+        ----------
+        metric : GraphSpaceMetric
+        base_graph : array-like, shape=[..., n_nodes, n_nodes]
+            Base graph.
+        graph_to_permute : array-like, shape=[..., n_nodes, n_nodes]
+            Graph to align.
+
+        Returns
+        -------
+        permuted_graph : array-like, shape=[..., n_nodes, n_nodes]
+            Permuted graph as to be aligned with respect to the geodesic.
+        """
+        self._set_all_perms(metric)
+
+        base_graph, graph_to_permute, is_single = self._broadcast(
+            base_graph, graph_to_permute
+        )
+        perms = [
+            self._align_single(metric, base_graph_, graph_to_permute_)
+            for base_graph_, graph_to_permute_ in zip(base_graph, graph_to_permute)
+        ]
+
+        self.perm_ = gs.array(perms[0]) if is_single else gs.array(perms)
+
+        return self._permute(metric, graph_to_permute, self.perm_)
+
+
+class _BasePointToGeodesicAligner(metaclass=ABCMeta):
+    """Base class for point to geodesic aligner.
+
+    Attributes
+    ----------
+    perm_ : array-like, shape=[...,n]
+        Node permutations where in position i we have the value j meaning
+        the node i should be permuted with node j.
+    """
+
+    def __init__(self):
+        self.perm_ = None
+
+    @abstractmethod
+    def align(self, geodesic, x):
+        raise NotImplementedError("Not implemented")
+
+    @abstractmethod
+    def dist(self, geodesic, x):
+        raise NotImplementedError("Not implemented")
+
+    def _get_n_points(self, x):
+        return 1 if gs.ndim(x) == 2 else gs.shape(x)[0]
+
+    def _permute(self, metric, graph_to_permute, perm):
+        return metric.space.permute(graph_to_permute, perm)
+
+
+class PointToGeodesicAligner(_BasePointToGeodesicAligner):
+    r"""Class for the Alignment of the points with respect to a geodesic.
+
+    Implementing the algorithm in [Huckemann2010] to select an optimal alignment to a
+    point with respect to a geodesic. The algorithm sample discrete set of n_points
+    along the geodesic between [s_min, s_max] and find the permutation that gets closer
+    to the datapoints along the geodesic.
+
+    Parameters
+    ----------
+    metric : GraphSpaceMetric
```

Comment:
Need parametrrs' descriptsion

Reply:
Especially because `s` is not a great variable name: it seems to refer to a convention used in a paper (which users might not read in details) rather than the mathematical concept it is representing. using `domain` would be more explicit